pkg/resource/permission/hooks.go

```diff
@@ -0,0 +1,330 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package permission
+
+import (
+	"context"
+	"strconv"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/ram"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws-controllers-k8s/ram-controller/apis/v1alpha1"
+)
+
+const (
+	StatusAttachable = "ATTACHABLE"
+)
+
+func (rm *resourceManager) customUpdatePermission(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+
+	if delta.DifferentAt("Spec.PolicyTemplate") {
+		err := rm.updatePermission(ctx, desired)
+		if err != nil {
+			return nil, err
+		}
+		ko.Status.Version = desired.ko.Status.Version
+		ko.Status.Status = desired.ko.Status.Status
+		ko.Status.LastUpdatedTime = desired.ko.Status.LastUpdatedTime
+		latest.ko.Status.Version = desired.ko.Status.Version
+		ackcondition.SetSynced(&resource{ko}, corev1.ConditionFalse, nil, nil)
+	}
+
+	if delta.DifferentAt("Spec.Tags") {
+		if err := rm.syncTags(ctx, desired, latest); err != nil {
+			return nil, err
+		}
+		ackcondition.SetSynced(&resource{ko}, corev1.ConditionTrue, nil, nil)
```

Comment:
If both `Spec.PolictyTemplate` and `Spec.Tags` have changed, this will set the resource as synced=true, which is not the expected behaviour, i believe? thoughts?